refactor(dashboard/user): profile view

CodeNamedRobin opened the pull request:
# Description
Refactors the entire profile view. Makes the profile view a lot neater, where no more cards are used for simple settings, instead all settings are in one user settings card.
## Related issues/external references
Many of #273
## Types of changes
<!-- What types of changes does your code introduce? Remove all the items that do not apply: -->
- New feature _(non-breaking change which adds functionality)_
- Style _(Change that do not affect the functionality of the code)_

CodeNamedRobin committed:
feat(dashboard/profile-view): started work on concept of user settings component

CodeNamedRobin committed:
feat(dashboard/user): Finalized outer look of user settings card

feat(dashboard/user): added change pin dialog to user settings card

CodeNamedRobin committed:
refactor(dashboard/user): removed change pin component

CodeNamedRobin committed:
feat(dashboard/user): added change pin form

CodeNamedRobin committed:
feat(dashboard/components): added FormSection component

feat(dashboard/user-profile): removed dialog for changing pin, added form section for changing pin

CodeNamedRobin committed:
feat(dashboard/profile-view): created ChangePasswordForm

CodeNamedRobin committed:
feat(dashboard/profile-view): added password change dialog

CodeNamedRobin committed:
feat(dashboard/profile-view): added delete api key and change api key links and confirmation dialogs

CodeNamedRobin committed:
feat(dashboard/profile-view): added extensive data processing logic to user settings component

CodeNamedRobin committed:
feat(dashboard/profile-view): added translations

CodeNamedRobin committed:
feat(dashboard/profile-view): added user info to profile view

SuperVK commented on apps/dashboard/src/modules/user/views/UserProfileView.vue:
```diff
@@ -1,51 +1,31 @@
-          <InputSwitch v-model="dataAnalysis" @update:modelValue="handleChange" />
-        </div>
-      </div>
+    <div class="flex md:flex-row flex-column-reverse justify-content-between gap-8">
```
I think gap-4 (or similiar) looks a lot better :)

CodeNamedRobin replied:
Has been implemented!

SuperVK reviewed (commented):
It would be nice if the buttons for pop up change to a cursor mouse on hover, and I think you might as well change the password changing to be the same as the pin change? Not sure tbh, but this feels weird while they kinda do the same

And do we want to implement Roy's suggestion while we are at it: #268

SuperVK commented:
Also I can't comment on the code, but the profile view somehow fetches all users and uses that to get your GEWIS id. But this endpoint will soon become privileged so it might be nice to do this differently (besides it already being very cursed that we fetch _all_ users to just use one)

CodeNamedRobin committed:
fix(dashboard/user-profile-view): max length for pincode showing dollar sign

fix(dashboard/user-profile-view): cursor not showing pointer on change api key and change password buttons

CodeNamedRobin committed:
feat(common/userstore): Added fetchGewisUser method

CodeNamedRobin commented:
> Also I can't comment on the code, but the profile view somehow fetches all users and uses that to get your GEWIS id. But this endpoint will soon become privileged so it might be nice to do this differently (besides it already being very cursed that we fetch _all_ users to just use one)

A new get individual gewis user fetch function has been implemented in the user store now.

CodeNamedRobin committed:
fix(dashboard/user-profile-view): type-check errors

SuperVK committed:
fix(profile-view): Better typing
